README.md

```diff
@@ -1 +1,123 @@
+
+## Configuration
+
+### `$BP_COMPOSER_VERSION`
+
+Use `$BP_COMPOSER_VERSION` to specify a version or range for Composer.
+When not specified, the default version will be used, typically the latest. 
+
+Any valid semver range is accepted.
+
+```shell
+BP_COMPOSER_VERSION=2.2.*
+```
+
+### `$BP_PHP_VERSION`
+
+Use `$BP_PHP_VERSION` to specify a version or range for the PHP requirement.
+When specified, it will take priority over the versions specified in either `composer.lock` or `composer.json` files. 
+When not specified, the default version will be used, typically the latest.
+
+Any valid semver range is accepted.
+
+```shell
+BP_PHP_VERSION=7.4.*
+```
+
+### `$BP_COMPOSER_INSTALL_OPTIONS`
+
+Use `$BP_COMPOSER_INSTALL_OPTIONS` to specify options for the Composer [install command](https://getcomposer.org/doc/03-cli.md#install-i).
+This buildpacks will always prepend `--no-progress` to the list of install options.
+The default is `--no-dev`.
+
+```shell
+# Note that env variables will typically be provided to this buildpack using `pack build -e`
+BP_COMPOSER_INSTALL_OPTIONS=--prefer-install --ignore-platform-reqs
+# will result in an installation command of `composer install --no-progress --prefer-install --ignore-platform-reqs`
+BP_COMPOSER_INSTALL_OPTIONS= # Note that this is set to empty
+# will result in an installation command of `composer install --no-progress`
+unset BP_COMPOSER_INSTALL_OPTIONS
+# will result in an installation command of `composer install --no-progress --no-dev`
+```
+
+### `$COMPOSER`
+
+The `$COMPOSER` variable allows you to specify the filename of `composer.json`.
+When set, this buildpack will use this location instead of `composer.json` in the detection phase. 
+This value must be relative to the project root. 
+
+For more information, please reference the [composer docs](https://getcomposer.org/doc/03-cli.md#composer).
+
+```shell
+COMPOSER=./somewhere/composer-other.json
+```
+
+## TODO:
+- Add layer caching
+- Offline caching
+- Add SBOM support
+- Add "Composer Global" support
+- Custom vendor directory via `$COMPOSER_VENDOR_DIR`
+- Check `composer.lock` and `composer.json` for appropriate PHP versions
+  - Note that Composer does not actually install PHP
+  - https://getcomposer.org/doc/01-basic-usage.md#platform-packages
+  - I'm not sure that this version is enforced by any part of the toolchain. 
+  - The `php-dist` buildpack only provides 64bit PHP, yet the default `php` dependency in Composer is 32bit
+  - What happens if `$COMPOSER` is present? Can we expect `composer.lock` to be a sibling of `composer.json`?
+  - It's perfectly valid to have a `composer.json`/`composer.lock` with no PHP dependency
+  - The `php-dist` buildpack only seems to provide 64 bit PHP. Is it a problem if this buildpack only asks for 32 bit?
+    - This would be true IFF composer.json has a dependency on `php` and not `php-64bit`
```

Comment:
@joshuatcasey can you elaborate on this a bit? Also tagging @dmikusa-pivotal - just wondering if you have any context on this

Reply:
There quite a lot of context on this one. It's also really relevant to above where this buildpack looks for `BP_PHP_VERSION`.

The `php-composer` buildpack inspects all three of: `BP_PHP_VERSION`, `composer.lock`, and `composer.json` to decide which PHP version to request. 

As it so happens, finding the right version of PHP that composer wants is a little tricky. If you find the `php` requirement that's explicitly the 32bit version, as opposed to `php-64bit`. The `php-composer` buildpack only inspects `php` but there was enough uncertainty here in my mind that I decided to punt on it for this PR.

https://getcomposer.org/doc/articles/composer-platform-dependencies.md

Reply:
Can you file an issue here to look into this in the future?

Reply:
Sure - could you clarify the issue? To determine whether the old `php-composer` method or the method proposed in this PR are valid ways to determine the desired PHP version?

Reply:
I think to investigate whether what we have in this PR is sufficient or if we need to move to the old method for completeness, yes! The solution might be to do nothing until we get feedback, but it doesn't hurt to have an issue.

Reply:
https://github.com/paketo-buildpacks/composer-install/issues/6